app/services/regions_integrity_check.rb

```diff
@@ -0,0 +1,138 @@
+class RegionsIntegrityCheck
+  SENTRY_ERROR_TITLE = "Regions Integrity Failure"
+
+  def self.sweep
+    new.sweep
+  end
+
+  attr_reader :errors
+
+  def initialize
+    @errors = {}
+  end
+
+  def sweep
+    resources.each do |name, resource|
+      log(:info, "Sweeping for – #{name}")
+      result = Result.check(resource)
+      errors[name] = result.inconsistencies unless result.ok?
+    end
+
+    report_errors
+    self
+  end
+
+  private
+
+  def resources
+    {
+      organizations: organizations,
+      facilities: facilities,
+      facility_groups: facility_groups,
+      blocks: blocks,
+      states: states
+    }
+  end
+
+  def organizations
+    {
+      source: Organization.pluck(:id),
+      region: Region.organization_regions.pluck(:source_id)
+    }
+  end
+
+  def facilities
+    {
+      source: Facility.pluck(:id),
```

Comment:
How do we need to use `with_discarded` for these integrity checks?

Reply:
Nah, I think neither `RegionBackfill` nor callbacks care about discarded records, so it wouldn't matter if they went out of sync anyway. I **think** it's ok to not have it.

Reply:
Does deleting facilities delete the region too? I'm thinking if we delete a facility, will we get a "region without source" show up in this check?

Reply:
[RegionSource](https://github.com/simpledotorg/simple-server/blob/master/app/models/region_source.rb#L4) handles this.